test/ssttpcalculator/CalculatorControllerSpec.scala

```diff
@@ -0,0 +1,162 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ssttpcalculator
+
+import akka.util.Timeout
+import journey.Statuses.InProgress
+import journey.{Journey, JourneyId, JourneyService}
+import org.scalatest.time.{Seconds, Span}
+import org.scalatestplus.play.PlaySpec
+import org.scalatestplus.play.guice.GuiceOneAppPerTest
+import play.api.Application
+import play.api.inject.guice.GuiceApplicationBuilder
+import play.api.http.Status
+import play.api.test.FakeRequest
+import play.api.test.Helpers.{redirectLocation, status}
+import ssttpaffordability.model.Expense.HousingExp
+import ssttpaffordability.model.{Expenses, Income, IncomeBudgetLine, Spending}
+import ssttpaffordability.model.IncomeCategory.MonthlyIncome
+import testsupport.RichMatchers.eventually
+import testsupport.stubs.{ArrangementStub, AuthStub, DirectDebitStub, TaxpayerStub}
+import testsupport.testdata.{TdAll, TdRequest}
+import testsupport.{RichMatchers, WireMockSupport}
+import uk.gov.hmrc.http.SessionKeys
+import uk.gov.hmrc.selfservicetimetopay.models.{BankDetails, EligibilityStatus, PaymentDayOfMonth, TypeOfAccountDetails}
+import _root_.model.enumsforforms.TypesOfBankAccount.Personal
+import _root_.model.enumsforforms.TypesOfBankAccount
+import org.scalatest.time.SpanSugar.convertIntToGrainOfTime
+import play.api.mvc.Result
+
+import java.time.LocalDateTime
+import java.util.UUID
+import scala.concurrent.Future
+
+class CalculatorControllerMaxLengthPlan12 extends CalculatorControllerSpec {
+  override def configuredMaxLengthOfPaymentPlan: Int = 12
+
+  override protected def createJourneyWithMaxLengthPlan(journeyId: JourneyId): Journey = {
+    createBaseJourney(journeyId)
+      .copy(maybeIncome = Some(Income(IncomeBudgetLine(MonthlyIncome, 1020))))
+      .copy(maybeSpending = Some(Spending(Expenses(HousingExp, 200))))
+  }
+
+  override protected def createJourneyNoAffordablePlan(journeyId: JourneyId): Journey = {
+    createBaseJourney(journeyId)
+      .copy(maybeIncome = Some(Income(IncomeBudgetLine(MonthlyIncome, 1000))))
+      .copy(maybeSpending = Some(Spending(Expenses(HousingExp, 200))))
+  }
+}
+
+class CalculatorControllerSpec extends PlaySpec with GuiceOneAppPerTest with WireMockSupport {
+
+  def configuredMaxLengthOfPaymentPlan: Int = 24
+
+  import TdRequest._
+
+  implicit val timeout: Timeout = Timeout(5.seconds)
+
+  val testPort: Int = 19001
+
+  val overrideConfig: Map[String, Any] = Map(
+    "calculatorType" -> CalculatorType.Legacy.value,
+    "legacyCalculatorConfig.maximumLengthOfPaymentPlan" -> configuredMaxLengthOfPaymentPlan
+  )
+
+  protected lazy val configMap: Map[String, Any] = Map(
```

Comment:
would it not work to extend this class by `ItSpec` and have these values picked up from there?

Reply:
Done. Thanks